src/block/block_addr.c

```diff
@@ -113,7 +113,7 @@ __wt_block_addr_pack(WT_BLOCK *block, uint8_t **pp, uint32_t objectid, wt_off_t
      * starting with a single object with no object IDs, where all future objects in the stack know
      * a missing object ID is a reference to the base object.
      */
-    if (i != 0 && block->has_objects) {
+    if (i != WT_BLOCK_OBJECTID_NONE) {
```

Comment:
It seems like this could be useful anywhere object ids are used. I suggest naming it `WT_TIERED_OBJECTID_NONE` and putting it in `tiered.h` instead. There is a line in `tiered_handle.c` that checks `tiered->current_id != 0` too.

Reply:
Fixed in c03722407.